lib/routes/vice/topic.ts

```diff
@@ -0,0 +1,110 @@
+    const response = await ofetch(`https://www.vice.com/${language}/topic/${topic}`);
+    const $ = load(response);
+    const nextData = JSON.parse($('script#__NEXT_DATA__').text());
+
+    const list = nextData.props.pageProps.listPageData.articles.map((item) => ({
+        title: item.title,
+        link: `https://vice.com${item.url}`,
+        pubDate: parseDate(item.publish_date, 'x'),
+        author: item.contributions.map((c) => c.contributor.full_name).join(', '),
+        description: item.dek,
+        category: [...new Set([item.primary_topic.name, ...item.topics.map((t) => t.name)])],
+    }));
+
+    const items = await Promise.all(
+        list.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const response = await ofetch(item.link);
+                const $ = load(response);
+                const articleNextData = JSON.parse($('script#__NEXT_DATA__').text()).props.pageProps.data.articles[0];
+                const bodyComponent = JSON.parse(articleNextData.body_components_json);
+
+                item.description =
+                    render({
+                        image: {
+                            url: articleNextData.thumbnail_url,
+                            alt: articleNextData.caption,
+                            caption: articleNextData.caption,
+                            credit: articleNextData.credit,
+                        },
+                    }) +
+                    bodyComponent
+                        .map((component) => {
+                            switch (component.role) {
+                                case 'body':
+                                    return render({ body: { html: component.html } });
+                                case 'heading2':
+                                    return render({ heading2: { html: component.html } });
+                                case 'divider':
+                                    return '';
+                                case 'image':
+                                    return render({
+                                        image: {
+                                            url: component.URL,
+                                            alt: component.alt,
+                                            caption: component.caption,
+                                        },
+                                    });
+                                case 'oembed':
+                                    return render({ oembed: { html: component.oembed.html } });
+                                case 'tweet':
+                                    return render({ oembed: { html: component.oembed.html } });
+                                case 'youtube':
+                                    return render({ oembed: { html: component.oembed.html } });
+                                case 'article':
+                                    return render({ article: { html: component.article.body } });
```

Comment:
`article` role provide some relative articles of the current article. I personally don't like having those's full articles embeded in the current article.

Reply:
I see what you are saying with this one. I was mainly doing it for news stories that essentially just contained a list of other articles. I didn't realise the mess it was making inside the other articles. I have now removed it in my latest commit. Thanks again for the pointers.